include/lsst/afw/detection/Psf.h

```diff
@@ -283,13 +298,29 @@ class Psf : public daf::base::Citizen,
     virtual geom::Box2I doComputeBBox(geom::Point2D const& position, image::Color const& color) const = 0;
     //@}
 
+    struct CacheKey {
+        geom::Point2D position;
+        image::Color color;
+
+        CacheKey(geom::Point2D const& position_, image::Color color_=image::Color())
+          : position(position_), color(color_) {}
+
+        friend std::size_t hash_value(CacheKey const& key) {
+            return boost::hash<geom::Point2D>()(key.position);  // Currently don't care about color
+        }
+
+        bool operator==(CacheKey const& other) const {
+            return position == other.position;  // Currently don't care about color
+        }
+
+        friend std::ostream & operator<<(std::ostream &os, CacheKey const &key) {
+            return os << key.position;
+        }
+    };
+
     bool const _isFixed;
-    mutable std::shared_ptr<Image> _cachedImage;
-    mutable std::shared_ptr<Image> _cachedKernelImage;
-    mutable image::Color _cachedImageColor;
-    mutable image::Color _cachedKernelImageColor;
-    mutable geom::Point2D _cachedImagePosition;
-    mutable geom::Point2D _cachedKernelImagePosition;
+    mutable utils::Cache<CacheKey, std::shared_ptr<Image>> _imageCache;
+    mutable utils::Cache<CacheKey, std::shared_ptr<Image>> _kernelImageCache;
```

Comment:
Maybe `shared_ptr<Image const>`, because it can be?

Reply:
Don't think I can, unless I cast away the same `const` in `Psf::computeImage`.